src/webgpu/shader/validation/expression/call/builtin/distance.spec.ts

```diff
@@ -57,32 +45,22 @@ Validates that constant evaluation and override evaluation of ${builtin}() never
     }
   })
   .fn(t => {
-    let expectedResult = true;
-
     const scalarType = scalarTypeOf(kValidArgumentTypes[t.params.type]);
-    const quantizeFn = quantizeFunctionForScalarType(scalarType);
+    const vCheck = new ConstantOrOverrideValueChecker(t, scalarType);
 
     // Distance equation: length(a - b)
     // Should be invalid if the calculations result in intermediate values that
     // exceed the maximum representable float value for the given type.
     const a = Number(t.params.a);
     const b = Number(t.params.b);
-    const ab = quantizeFn(a - b);
-
-    if (!Number.isFinite(ab)) {
-      expectedResult = false;
-    }
+    const ab = vCheck.checkedResult(a - b);
 
     // Only calculates the full length if the type is a vector. Otherwise abs(a-b) is used.
     if (kValidArgumentTypes[t.params.type].width > 1) {
-      const ab2 = quantizeFn(ab * ab);
-      const sqrLen = quantizeFn(ab2 * kValidArgumentTypes[t.params.type].width);
+      const ab2 = vCheck.checkedResult(ab * ab);
+      const sqrLen = vCheck.checkedResult(ab2 * kValidArgumentTypes[t.params.type].width);
       // Square root does not need to be calculated because it can never fail if
```

Comment:
I don't think this comment is correct...
sqrt's accuracy is defined as 1/inverseSqrt(x)... which becomes a pain, because the inteval for inverseSqrt is 2 ULP, and division is 2.5 ULP plus not being define very close to 0 (https://www.w3.org/TR/WGSL/#floating-point-accuracy) So there should probably be a check that the value going into sqrt isn't near 0... and if it is, just skip, because it is likely to be a may fail case.

Reply:
Thanks for pointing that out! Updated to add a near-zero check, implemented by testing to see if the value is between the min positive and max negative values for the type.